Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,587 @@
+            )
+            .getCodeExtraInformation()
+            .setFunctionName(`get${property.functionName}`);
+        }
+
+        // Add all the generic setters
+        if (parameterType === 'number' || parameterType === 'string') {
+          const expressionType =
+            parameterType === 'number' ? 'expression' : 'string';
+          gdObject
+            .addAction(
+              `Set${property.functionName}`,
+              _('Set ') + property.description,
+              _('Set ') + property.description,
+              _('Do _PARAM1__PARAM2_ to the ') +
+                property.description +
+                _(' of _PARAM0_'),
+              '',
+              property.iconPath,
+              property.iconPath
+            )
+            .addParameter(
+              'object',
+              objectName + _(' object'),
+              objectName,
+              false
+            )
+            .addParameter('operator', _("Modification's sign"), '', false)
+            .addParameter(expressionType, property.paramLabel, '', false)
+            .getCodeExtraInformation()
+            .setFunctionName(`set${property.functionName}`)
+            .setManipulatedType(parameterType)
+            .setGetter(`get${property.functionName}`);
+        } else {
+          // Setter doesnt have a +- modifier (Color, alignment, etc)
+          gdObject
+            .addAction(
+              `Set${property.functionName}`,
+              _('Set ') + property.description,
+              _('Set ') + property.description,
+              _('Set ') + property.description + _(' of _PARAM0_ to _PARAM1_'),
+              '',
+              property.iconPath,
+              property.iconPath
+            )
+            .addParameter(
+              'object',
+              objectName + _(' object'),
+              objectName,
+              false
+            )
+            .addParameter(
+              parameterType,
+              property.paramLabel,
+              property.options ? property.options : '',
+              false
+            )
+            .getCodeExtraInformation()
+            .setFunctionName(`set${property.functionName}`)
+            .setGetter(`get${property.functionName}`);
+        }
+
+        // Add compare Conditions via getters
+        if (parameterType === 'string' || parameterType === 'number') {
+          const propExpressionType =
+            parameterType === 'string' ? 'string' : 'expression';
+          gdObject
+            .addCondition(
+              `Is${property.functionName}`,
+              property.description,
+              _('Compare the value of ') + property.description,
+              _('The ') +
+                property.paramLabel +
+                _(' of _PARAM0_ is _PARAM1__PARAM2_'),
+              '',
+              property.iconPath,
+              property.iconPath
+            )
+            .addParameter(
+              'object',
+              objectName + _(' object'),
```

Comment:
We need to avoid string concatenation for translations because this result in bad quality translations (translator forgot to add leading/trailing space) or wrong word order in some languages.
Here, it's the description shown to the user - which actually won't be shown using the new editor. Just call it `_("Object")` (or `objectFullName` if you really want to, see next comment)